src/common/transport.cc

```diff
@@ -200,6 +200,7 @@ Transport::asyncWriteImpl(
         if (bytesWritten < 0) {
             if (errno == EAGAIN || errno == EWOULDBLOCK) {
                 if (status == FirstTry) {
+                    toWrite.erase(fd);
```

Comment:
Could you please explain why this `erase` is needed ? It's not clear to me.

Basically, if it's the first time we're hitting `EAGAIN`, the fd should not be in the map. Maybe is it not cleaned afterwise ?